kimchi/src/circuits/polynomials/lookup.rs

```diff
@@ -661,16 +635,26 @@ pub fn constraints<F: FftField>(configuration: &LookupConfiguration<F>, d1: D<F>
             .collect()
     };
 
+    let eval = |pos: LocalPosition| {
+        E::Cell(Variable {
+            col: Column::Witness(pos.column),
+            row: pos.row,
+        })
+    };
+
     // This is set up so that on rows that have lookups, chunk will be equal
     // to the product over all lookups `f` in that row of `gamma + f`
     // and
     // on non-lookup rows, will be equal to 1.
-    let f_term = |spec: &Vec<_>| {
+    let f_term = |spec: &Vec<JointLookupSpec<_>>| {
         assert!(spec.len() <= lookup_info.max_per_row);
         let padding = complements_with_beta_term[lookup_info.max_per_row - spec.len()].clone();
 
         spec.iter()
-            .map(|j| E::Constant(ConstantExpr::Gamma) + joint_lookup(j))
+            .map(|j| {
+                E::Constant(ConstantExpr::Gamma)
+                    + j.evaluate(E::constant(ConstantExpr::JointCombiner), &eval)
```

Comment:
I'm confused as to how you can pass a reference to a closure to evaluate, which doesn't seem to expect a closure there

Reply:
I think rust-analyzer is getting confused as to which `evaluate()` function is called o_O

Reply:
https://github.com/rust-analyzer/rust-analyzer/issues/11759

Reply:
Commented there; presumably it's due to name reuse.